esp-hal/src/gpio/mod.rs

```diff
@@ -1238,8 +1317,47 @@ impl<'d, P> Peripheral for Input<'d, P> {
 }
 
 impl<'d> Input<'d> {
-    /// Create GPIO input driver for a [Pin] with the provided [Pull]
-    /// configuration.
+    /// Creates a new, type-erased GPIO input.
+    ///
+    /// The `pull` parameter configures internal pull-up or pull-down
+    /// resistors.
+    ///
+    /// ## Example
+    ///
+    /// The following example configures `GPIO5` to read a button press. The
+    /// example assumes that the button is connected such that the pin is low
+    /// when the button is pressed.
+    ///
+    /// ```rust, no_run
+    #[doc = crate::before_snippet!()]
+    /// use esp_hal::gpio::{Level, Input, Pull};
+    /// use esp_hal::delay::Delay;
+    ///
+    /// # macro_rules! println {
```

Comment:
Maybe we can define this macro in `before_snippet` to have it available everywhere?

Reply:
I guess we can!